### DISCUSSION TITLE
Opening Octo Buffers Outside of Octo, e.g., with `:edit` or Harpoon

### DISCUSSION THREAD

**Discussion by paulhdk** [pwntester/octo.nvim#649] [General]
[The harpoon plugin](https://github.com/ThePrimeagen/harpoon/tree/harpoon2) allows you to pin files in a picker, which persists across Neovim restarts.

I've noticed that after a restart, opening octo buffers through harpoon - or even just with `:edit octo://...` some syntax highlighting is missing, e.g., the diff bar and the assignees aren't coloured.

If I open the Octo picker with the partially coloured PR open and close it again, everything is coloured correctly.
So, I assume opening Octo - not just loading it via Lazy - triggers some additional functionality that I'm not calling when opening an Octo PR without going through Octo itself.

Does anyone have experience using octo with harpoon or have some pointers how I could address this?

**Comment by williambdean** [comment]
I enter the octo files various ways and have noted some lack of rendering on occasion. Not sure exactly the reason. I haven't noticed any issues with harpoon.

Can you post a screenshot of what you mean?

My understanding is that it should be based on the auto-commands defined [here](https://github.com/pwntester/octo.nvim/blob/master/lua/octo/autocmds.lua) in case you want to do some digging

**Reply by paulhdk** [reply]
> I enter the octo files various ways and have noted some lack of rendering on occasion. Not sure exactly the reason. I haven't noticed any issues with harpoon.

Even after restarting Neovim and then accessing an Octo file with harpoon?

> Can you post a screenshot of what you mean?

Full colouring
<img width="2560" alt="Screenshot 2024-10-25 at 09 48 45" src="https://github.com/user-attachments/assets/4cbe5436-3d55-4e06-8ee1-956a16a4770d">


Partial colouring, e.g., after restarting Neovim and acceessing the Octo file through harpoon:
<img width="2560" alt="Screenshot 2024-10-25 at 09 49 04" src="https://github.com/user-attachments/assets/f6968ddf-456b-4539-a213-025f84657749">

> My understanding is that it should be based on the auto-commands defined [here](https://github.com/pwntester/octo.nvim/blob/master/lua/octo/autocmds.lua) in case you want to do some digging

I’ll have a look! Simply calling `:lua require(“octo”).configure_octo_buffer()` didn’t work.

**Reply by williambdean** [reply]
I experience this as well. I am thinking the highlight groups arent initialized or something. 
Have you played around with it more and discovered anything?

**Reply by williambdean** [reply]
I am going to make an issue for this instead of discussion

**Reply by williambdean** [reply]
The issue is #771
